src/test/java/integ/global_attribute/GlobalAttributeProcess.java

```diff
@@ -0,0 +1,213 @@
+import static integ.global_attribute.GlobalAttributeProcess.PK_KEY;
+import static integ.global_attribute.GlobalAttributeProcess.PK_VALUE;
+import static integ.global_attribute.GlobalAttributeProcess.TABLE_NAME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import io.xdb.core.command.CommandRequest;
+import io.xdb.core.command.CommandResults;
+import io.xdb.core.communication.Communication;
+import io.xdb.core.context.Context;
+import io.xdb.core.exception.persistence.GlobalAttributeNotFoundException;
+import io.xdb.core.persistence.Persistence;
+import io.xdb.core.persistence.PersistenceTableRowToUpsert;
+import io.xdb.core.persistence.schema.PersistenceSchema;
+import io.xdb.core.persistence.schema.PersistenceTableColumnSchema;
+import io.xdb.core.persistence.schema.PersistenceTableSchema;
+import io.xdb.core.persistence.schema_to_load.PersistenceSchemaToLoad;
+import io.xdb.core.persistence.schema_to_load.PersistenceTableSchemaToLoad;
+import io.xdb.core.process.Process;
+import io.xdb.core.process.ProcessOptions;
+import io.xdb.core.process.ProcessStartConfig;
+import io.xdb.core.state.AsyncState;
+import io.xdb.core.state.AsyncStateOptions;
+import io.xdb.core.state.StateDecision;
+import io.xdb.core.state.StateSchema;
+import io.xdb.gen.models.AttributeWriteConflictMode;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GlobalAttributeProcess implements Process {
+
+    public static final String TABLE_NAME = "sample_user_table";
+    public static final String PK_KEY = "user_id";
+    public static final String PK_VALUE = "pk_value";
+    public static final String COL_KEY_1 = "first_name";
+    public static final String COL_VALUE_1 = "col_1_value";
+    public static final String COL_VALUE_1_2 = "col_1_value_2";
+    public static final String COL_KEY_2 = "last_name";
+    public static final String COL_VALUE_2 = "col_2_value";
+    public static final String COL_KEY_3 = "create_timestamp";
+    public static final int COL_VALUE_3 = 111;
+
+    @Override
+    public StateSchema getStateSchema() {
+        return StateSchema.withStartingState(
+            new GlobalAttributeProcessStartingState(),
+            new GlobalAttributeProcessNextState1(),
+            new GlobalAttributeProcessNextState2()
+        );
+    }
+
+    @Override
+    public PersistenceSchema getPersistenceSchema() {
+        return PersistenceSchema.withGlobalAttributes(
+            PersistenceTableSchema
+                .withSingleColumnPrimaryKey(TABLE_NAME, PersistenceTableColumnSchema.create(PK_KEY, String.class, true))
+                .addColumn(PersistenceTableColumnSchema.create(COL_KEY_1, String.class, true))
+                .addColumn(PersistenceTableColumnSchema.create(COL_KEY_2, String.class, true))
+                .addColumn(PersistenceTableColumnSchema.create(COL_KEY_3, Integer.class, false))
+        );
```

Comment:
Comparing this with the golang sdk approach, I still feel like the golang one is more readable for me:
https://github.com/xdblab/xdb-golang-samples/blob/5c657cd0a058c5bf7a91e12a65d02d989930bcdc/processes/signup/process.go#L40

I would keep them similar:
```
PersistenceSchema.create(
     LocalAttributeSchema.empty(),
     GlobalAttributeSchema.create(
           DBTableSchema.create(
               TABLE_NAME, 
               DBPrimaryKeyDef.create(PK_KEY, String.class, true),
               DBColumnDef.create(...),
               DBColumnDef.create(...),
           ),
     ),
);
```

Based on that we can have `DBPrimaryKeyDef.createForMultiColumns(...),

And add an overloaded parameter for the named persistencePolicy.

Reply:
Or change "create" to "def" to be shorter, something like 
```
PersistenceSchema.def(
     LocalAttributeSchema.empty(),
     GlobalAttributeSchema.def(
           DBTableSchema.def(
               TABLE_NAME, 
               DBPrimaryKey.def(PK_KEY, String.class, true),
               DBColumn.def("user_id", String.class, true),
               DBColumn.def(...),
           ),
     ),
);
```

Reply:
I think Java's way is more on the current implemented style. E.g.., the builders are like `XXX.builder().setXXX().setXXX()`.

Reply:
Agreed that `def` is too strange. I would take that back.

I don't think there is a Java pattern like `Schema.withSingleColumnPrimaryKey(...)`. 

There are two main patterns:
1. Just use constructor/constructor helper  like Schema.create(p1, p2, p3) as I suggested.
2. JavaBean Pattern
3. Builder pattern.

The Java bean pattern is :
```
schema = new Schema()
schema.setP1(p1)
schema.setP2(p2)
...
```

The builder pattern is :
```
schema = Scheam.builder()
.p1(p1)
.p2(p2)
p3(p3)
.build();
```
(build pattern doesn't need setXXX)

My suggestion is that unless the list of parameter is too huge, I would prefer to just use constructor with helper.

PersistenceSchema.create( localAttributeSchema, globalAttributeSchema, [namedPersistencePolicies]). 

User can put `LocalAttributeScheam.empty()` for the first parameter to explicitly tell it is empty. 

The builder pattern will be necessary when there are a big list of parameter to use. 
As said in the StackOverflow:

https://stackoverflow.com/questions/328496/when-would-you-use-the-builder-pattern

> The problem with this pattern is that once constructors are 4 or 5 parameters long it becomes difficult to remember 

I am not against to builder pattern because there are 3 parameters already, closed to 4/5. It's still not necessary. We could provide the builder pattern later.

Reply:
Actually, java's grpc request/response pattern is what I described here: https://github.com/xdblab/xdb-java-sdk/pull/12#discussion_r1392657004.

Reply:
The current way is more like the builder pattern, with the builder renamed as withXXX. I feel now in Java SDK there is mixed usage of different naming styles from Golang and Java, we might revisit them all at once to define a consistent style for Java SDK, and removing the preference from golang.

Reply:
> The current way is more like the builder pattern, with the builder renamed as withXXX. 

I don't think it's even closed...

The GRPC request/response builder is like this 
```
HelloReply.newBuilder().setMessage("Hello again " + req.getName()).build();
```
see https://grpc.io/docs/languages/java/quickstart/

As a builder pattern, you need to have a builder object/static inner class. 

(and yes, GRPC request/response builder do use `setXXX` which I am not sure why, they may have other reasons, but I don't think it's necessary for us to follow if we use builder pattern. 

> there is mixed usage of different naming styles from Golang and Java
The builder pattern is not particular for Java, Golang can do the same way. but it's just more popular in Java. 

>  removing the preference from golang

I am not trying to follow Golang for Java, and again I am not against to builder pattern.  I just don't think it's necessary to build builder pattern if there are less than 5 things to build into an object.  

Otherwise, if we just use builder pattern for everything "consistently", this is what you will see instead:
```java
PersistenceSchema.newBuilder()
     .localAttributeSchema( ...)
     .globalAttributeSchema(
          GlobalAttributeSchema.newBuilder()
                 .addDBTableSchema(
                         DBTableSchema.newBuilder()
                                .tableName(TBL_NAME)
                                .addPrimaryKeyDef( 
                                       PrimaryKeyDef.newBuilder()
                                            .keyName("id")
                                            .loadDefault(true)
                                            .type(String.class)
                                            .build
                                )
                                .addPrimaryKey( PrimaryKeyDef.newBuilder()....build())
                                .addColumnDef(
                                   ColumnDef.newBuilder()
                                        .columnName("username")
                                        .columnType(String.class)
                                        .loadDefault(true)
                                        .build()
                                )
                                .addColumnDef(ColumnDef.newBuilder()....build() )
                 )
                 .addDBTableSchema(...)
                 .build();
          )       
   ).build();

```

When I am writing the above, I am thinking that as a compromise, some of them can adopt the builder pattern:
like DBTableSchema as it has quite many fields. 
But others I think it's better to keep it simple to save the typing from users. WDYT?

Reply:
Going to use a mixed way because now the global attribtues have not been finalized (I still have concerns of mixing use attributes vs tables/columns), and I am not sure how the local attributes are going to be presented, so I will leave a geneal naming like `PersistenceTableSchema` for now. We can revisit the codes once the persistence is finalized. 

```
return PersistenceSchema.withGlobalAttributes(
            PersistenceTableSchema.create(
                TABLE_NAME,
                PersistenceTableColumnSchema.create(PK_KEY, String.class, true, true),
                PersistenceTableColumnSchema.create(COL_KEY_1, String.class, false, true),
                PersistenceTableColumnSchema.create(COL_KEY_2, String.class, false, true),
                PersistenceTableColumnSchema.create(COL_KEY_3, Integer.class, false, false)
            )
        );
```